RenderStateNotation/src/RenderStateNotationLoaderImpl.cpp

```diff
@@ -0,0 +1,395 @@
+/*
+ *  Copyright 2019-2022 Diligent Graphics LLC
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  In no event and under no legal theory, whether in tort (including negligence),
+ *  contract, or otherwise, unless required by applicable law (such as deliberate
+ *  and grossly negligent acts) or agreed to in writing, shall any Contributor be
+ *  liable for any damages, including any direct, indirect, special, incidental,
+ *  or consequential damages of any character arising as a result of this License or
+ *  out of the use or inability to use the software (including but not limited to damages
+ *  for loss of goodwill, work stoppage, computer failure or malfunction, or any and
+ *  all other commercial damages or losses), even if such Contributor has been advised
+ *  of the possibility of such damages.
+ */
+
+#include "RenderStateNotationLoaderImpl.hpp"
+#include "DefaultRawMemoryAllocator.hpp"
+
+namespace Diligent
+{
+RenderStateNotationLoaderImpl::RenderStateNotationLoaderImpl(IReferenceCounters* pRefCounters, const RenderStateNotationLoaderCreateInfo& CreateInfo) :
+    TBase{pRefCounters},
+    m_pDevice{CreateInfo.pDevice},
+    m_pParser{CreateInfo.pParser},
+    m_pStream{CreateInfo.pStream}
+{
+    VERIFY_EXPR(CreateInfo.pDevice != nullptr && CreateInfo.pParser != nullptr);
+}
+
+void RenderStateNotationLoaderImpl::LoadPipelineState(const LoadPipelineStateInfo& LoadInfo, IPipelineState** ppPSO)
+{
+    DEV_CHECK_ERR(LoadInfo.Name != nullptr, "LoadInfo.Name  must not be null");
+    DEV_CHECK_ERR(ppPSO != nullptr, "ppPSO must not be null");
+
+    try
+    {
+        auto Iter = m_PipelineStateCache.find(LoadInfo.Name);
+        if (Iter != m_PipelineStateCache.end())
+        {
+            *ppPSO = Iter->second;
+        }
+        else
+        {
+            DynamicLinearAllocator Allocator{DefaultRawMemoryAllocator::GetAllocator()};
+
+            auto FindShader = [&](const char* Name) -> IShader* {
```

Comment:
You are returning raw pointers, which is unsafe and makes object onwership unclear.
Return `RefCntAutoPtr<IShader>`.
There is also no need to explicitly define the lambda return type when it can be deduced from return value.